app/test/vesting_test.go

```diff
@@ -0,0 +1,507 @@
+	s.accountsMut.Lock()
+	defer s.accountsMut.Unlock()
+
+	accountsAny, found := s.accounts.Load(accType)
+	if !found {
+		return "", fmt.Errorf("account type `%s` not found", accType.String())
+	}
+
+	accounts := accountsAny.(accountDispenser)
+	if accounts.counter >= len(accounts.names) {
+		return "", fmt.Errorf("out of unused accounts for type `%s`", accType.String())
+	}
+
+	name := accounts.names[accounts.counter]
+	accounts.counter++
+	s.accounts.Store(accType, accounts)
+
+	return name, nil
+}
+
+// getAnUnusedContinuousVestingAccount returns an unused continuous vesting account and its name.
+//
+// It takes a minimum start-time as input and finds an unused account whose start time is greater than the input.
+func (s *VestingModuleTestSuite) getAnUnusedContinuousVestingAccount(minStartTime int64) (vAcc vestingtypes.ContinuousVestingAccount, name string, err error) {
+	for {
+		name, err = s.unusedAccount(ContinuousVestingAccountType)
+		if err != nil {
+			return vAcc, name, err
+		}
+		address := getAddress(name, s.cctx.Keyring).String()
+		resAccBytes, err := testfactory.GetRawAccountInfo(s.cctx.GRPCClient, address)
+		if err != nil {
+			return vAcc, name, err
+		}
+
+		err = vAcc.Unmarshal(resAccBytes)
+		if err != nil || vAcc.StartTime > minStartTime {
+			return vAcc, name, err
+		}
+	}
+}
+
+// getAnUnusedPeriodicVestingAccount returns an unused periodic vesting account and its name.
+//
+// It takes a minimum start-time as input and finds an unused account whose start time is greater than the input.
+func (s *VestingModuleTestSuite) getAnUnusedPeriodicVestingAccount(minStartTime int64) (vAcc vestingtypes.PeriodicVestingAccount, name string, err error) {
+	for {
+		name, err = s.unusedAccount(PeriodicVestingAccountType)
+		if err != nil {
+			return vAcc, name, err
+		}
+		address := getAddress(name, s.cctx.Keyring).String()
+		resAccBytes, err := testfactory.GetRawAccountInfo(s.cctx.GRPCClient, address)
+		if err != nil {
+			return vAcc, name, err
+		}
+
+		err = vAcc.Unmarshal(resAccBytes)
+		if err != nil || vAcc.StartTime > minStartTime {
+			return vAcc, name, err
+		}
+	}
+}
+
+// getAnUnusedDelayedVestingAccount returns the name of an unused delayed vesting account.
+//
+// It takes a minimum end-time as input and finds an unused account whose end time is greater than the input.
+func (s *VestingModuleTestSuite) getAnUnusedDelayedVestingAccount(minEndTime int64) (vAcc vestingtypes.DelayedVestingAccount, name string, err error) {
+	for {
+		name, err = s.unusedAccount(DelayedVestingAccountType)
+		if err != nil {
+			return vAcc, name, err
+		}
+
+		address := getAddress(name, s.cctx.Keyring).String()
+		resAccBytes, err := testfactory.GetRawAccountInfo(s.cctx.GRPCClient, address)
+		if err != nil {
+			return vAcc, name, err
+		}
+
+		err = vAcc.Unmarshal(resAccBytes)
+		if err != nil || vAcc.EndTime > minEndTime {
+			return vAcc, name, err
+		}
+	}
```

Comment:
is there any way to avoid using for loops for simply lookups?

Reply:
Since we have to set everything at genesis for these tests and it is time sensitive and we have no control over time, I could not come up with a loopless solution. Maybe if we have a paradigm shift and look at the problem from another angel, a solution could be found.